Fix #1429: Correct App Group id for Beta/TestFlight builds.

iccub opened the pull request:
<!-- *Thank you for submitting a pull request, your contributions are greatly appreciated!* -->

## Summary of Changes
1. We fix the app group id for beta builds
2. If it fails we will save profile folder to `Library/Application Support`

This pull request fixes issue #1429 
<!-- If no ticket has been fixed, please file one now: https://github.com/brave/brave-ios/issues -->

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:
<!-- Any useful notes explaining how best to test and verify. -->


## Screenshots:
<!-- If your patch includes user interface changes that you would like to suggest or that you would like UX to look at, please include them here. -->


## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/internal/issues/new?assignees=&labels=security%2C+security-reviews&template=security-review.md&title=iOS:%20Security+Review+for) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

iccub committed:
Fix #1429: Correct App Group id for Beta/TestFlight builds.

jhreis commented on Shared/AppInfo.swift:
```diff
@@ -39,10 +39,9 @@ open class AppInfo {
-            bundleIdentifier = "com.brave.ios.Fennec.enterprise"
+        if bundleIdentifier == "com.brave.ios.BrowserBeta" {
+            // App group identifier for beta builds is different from its bundle id.
+            bundleIdentifier = "com.brave.ios.BrowserBeta.unique"
```
Can we change the app group identifier? I'm guessing this was used (probably by me), due to the group id being taken already.

iccub replied:
Yes it should be possible

iccub commented:
I fixed the app group, now it is the same as our beta build id.

jhreis reviewed (changes requested):
Let's actually add a unit test to check if any files are created in document directory at launch. Basically:
- launch application
- wait 5-10 seconds
- verify documents directory is completely empty

iccub committed:
No Bug: Add unit test to verify that documents directory is empty.

This commit also moves other test files from documents directory
to application support.

iccub commented:
Added the test as separate commit, don't squash please :)
